### PR TITLE
fix(snowflake): transpile DuckDB JSON_ARRAY to ARRAY_CONSTRUCT [CLAUDE]

### PR DESCRIPTION
**Pull request by maxa-jonathan-hallee**
## Summary

DuckDB's `JSON_ARRAY` function has no Snowflake equivalent — Snowflake uses `ARRAY_CONSTRUCT` instead (`JSON_ARRAY` is not a recognized Snowflake function).

Currently, `JSON_ARRAY(...)` falls through to `Anonymous` in the DuckDB parser and passes through unchanged to Snowflake output, producing invalid SQL.

**Changes:**
- Add `JSON_ARRAY` → `exp.JSONArray` mapping in DuckDB parser `FUNCTIONS` dict (follows the SingleStore pattern at `parsers/singlestore.py:107`)
- Add `exp.JSONArray` → `ARRAY_CONSTRUCT` transform in Snowflake generator `TRANSFORMS` dict

**Before:**
```sql
-- DuckDB input
SELECT JSON_ARRAY('a', 'b', 'c')

-- Snowflake output (broken)
SELECT JSON_ARRAY('a', 'b', 'c')
```

**After:**
```sql
-- Snowflake output (correct)
SELECT ARRAY_CONSTRUCT('a', 'b', 'c')
```

**Reference:** [Snowflake ARRAY_CONSTRUCT docs](https://docs.snowflake.com/en/sql-reference/functions/array_construct)

## Test plan

- Added `validate_all` test for `JSON_ARRAY` → `ARRAY_CONSTRUCT` cross-dialect transpilation in `test_duckdb.py`
- All existing DuckDB tests pass (44 passed, 585 subtests)
- All existing Snowflake tests pass (95 passed, 1567 subtests)

### CODE REVIEW

File: tests/dialects/test_duckdb.py [hunk 1]
```diff
@@ -548,6 +548,13 @@ def test_duckdb(self):
             """SELECT JSON_EXTRACT_STRING('{ "family": "anatidae", "species": [ "duck", "goose", "swan", null ] }', ['$.family', '$.species'])""",
             """SELECT '{ "family": "anatidae", "species": [ "duck", "goose", "swan", null ] }' ->> ['$.family', '$.species']""",
         )
+        self.validate_all(
+            "SELECT JSON_ARRAY('a', 'b', 'c')",
+            write={
+                "duckdb": "SELECT JSON_ARRAY('a', 'b', 'c')",
+                "snowflake": "SELECT ARRAY_CONSTRUCT('a', 'b', 'c')",
+            },
```
Comment (geooo109) [inline tests/dialects/test_duckdb.py:577]: What happens in the case of `NULL` values ? 

Also, If the input duckdb query is something like:
```
 SELECT JSON_ARRAY('a', 1, 'c')->>1;
```
This approach, won't work right ? because we apply a `JSON` style access here, the transpiled output won't be valid. (because we return an array)

So, we should wrap the result with `TO_JSON`. (duckdb returns a json array, we have to match the semantics for the snowflake side)

Comment (maxa-jonathan-hallee) [reply tests/dialects/test_duckdb.py:577]: Yeah good point @geooo109 

I've added the TO_JSON wrapping + extra unit tests for NULL

File: sqlglot/generators/snowflake.py [hunk 2]
```diff
@@ -497,6 +497,9 @@ class SnowflakeGenerator(generator.Generator):
         exp.GetExtract: rename_func("GET"),
         exp.GroupConcat: lambda self, e: groupconcat_sql(self, e, sep=""),
         exp.If: if_sql(name="IFF", false_value="NULL"),
+        exp.JSONArray: lambda self, e: self.func(
+            "TO_JSON", self.func("ARRAY_CONSTRUCT", *e.expressions)
+        ),
```
Comment (georgesittas) [inline sqlglot/generators/snowflake.py:502]: Hey @maxa-jonathan-hallee, given your [previous PR](https://github.com/tobymao/sqlglot/pull/7496), which introduced this transpilation:

```
(sqlglot) ➜  sqlglot git:(main) sqlglot --read duckdb --write snowflake "SELECT x::JSON"
SELECT CAST(x AS VARIANT)
```

Isn't this transpilation inconsistent? Snowflake's `TO_JSON` function returns a `VARCHAR`, not a `VARIANT`. I believe we should use `TO_VARIANT`, instead.

For example, @geooo109's example will work like this:

```
(sqlglot) ➜  sqlglot git:(main) sqlglot --read duckdb --write snowflake "SELECT JSON_ARRAY('a', 'b') ->> 1;"
SELECT JSON_EXTRACT_PATH_TEXT(TO_VARIANT(ARRAY_CONSTRUCT('a', 'b')), '[1]')
```

Comment (geooo109) [reply sqlglot/generators/snowflake.py:502]: yeah, nice catch! `TO_VARIANT` is the correct approach here.

Comment (maxa-jonathan-hallee) [reply sqlglot/generators/snowflake.py:502]: Ah yeah my bad. Should be resolved now. Thanks to both of you
